pin-project-internal/src/pin_project/derive.rs

```diff
@@ -709,16 +918,37 @@ impl<'a> Context<'a> {
     }
 
     /// Creates an implementation of the projection method.
-    fn make_proj_impl(&self, proj_body: &TokenStream, proj_ref_body: &TokenStream) -> TokenStream {
+    fn make_proj_impl(
+        &self,
+        proj_body: &TokenStream,
+        proj_ref_body: &TokenStream,
+        proj_own_body: &TokenStream,
+    ) -> TokenStream {
         let vis = &self.proj.vis;
         let lifetime = &self.proj.lifetime;
         let orig_ident = self.orig.ident;
         let proj_ident = &self.proj.mut_ident;
         let proj_ref_ident = &self.proj.ref_ident;
+        let proj_own_ident = &self.proj.own_ident;
 
+        let orig_ty_generics = self.orig.generics.split_for_impl().1;
         let proj_ty_generics = self.proj.generics.split_for_impl().1;
         let (impl_generics, ty_generics, where_clause) = self.orig.generics.split_for_impl();
 
+        let replace_impl = self.replace.map(|replace| {
+            quote_spanned! { replace =>
+                #[allow(unsafe_code)]
```

Comment:
```suggestion
```

`#[deny(unsafe_code)]` does not deny unsafe code generated by macros (https://github.com/rust-lang/rust/issues/53975)

Reply:
Hm, I only added this because it failed the test suite without it, so it doesn't appear like that behaviour is working correctly.